# This patch is included in the core repository: https://github.com/MetaMask/core/pull/5355 and will most likely be released in version 46.0.1.
# To avoid applying version 46.0.0 and manage breaking changes, we need to apply this tiny patch.
# For this reason, there is no working branch for this patch in the core repository
diff --git a/dist/TransactionController.cjs b/dist/TransactionController.cjs
index 176acdb4dd711b5c956d3c82091075c0ea379316..c3038af72d505a4187389adb15ee46bde98519b6 100644
--- a/dist/TransactionController.cjs
+++ b/dist/TransactionController.cjs
@@ -1155,7 +1155,11 @@ class TransactionController extends base_controller_1.BaseController {
                 if (!isTxCompleted) {
                     if (error?.code === rpc_errors_1.errorCodes.provider.userRejectedRequest) {
                         this.cancelTransaction(transactionId, actionId);
-                        throw rpc_errors_1.providerErrors.userRejectedRequest('MetaMask Tx Signature: User denied transaction signature.');
+                        throw rpc_errors_1.providerErrors.userRejectedRequest({
+                            message:
+                                'MetaMask Tx Signature: User denied transaction signature.',
+                            data: error?.data,
+                        });
                     }
                     else {
                         this.failTransaction(meta, error, actionId);
